packages/composite-checkout/README.md

```diff
@@ -288,6 +289,10 @@ A React Hook that returns all the payment processor functions in a Record.
 
 A React Hook that will return the `onClick` function passed to each [payment method's submitButton component](#payment-methods). The hook requires a payment processor ID. Call the returned function with data for the payment processor and it will begin the transaction.
 
+### useMakeStepActive
+
+A React Hook that will return a function to set a step to be the active step. Only works within a step. The returned function looks like `( stepId: string ) => Promise< boolean >;`. Calling this function is similar to pressing the "Edit" button on the specified step. Note that this is risky! It will make the previous active step inactive, but **will not complete** that step, so if the previous step has any `isCompleteCallback` behavior, that behavior will not be triggered. It will also ignore the `canEditStep` prop of `CheckoutStep`.
```

Comment:
> but **will not complete** that step, so if the previous step has any `isCompleteCallback` behavior, that behavior will not be triggered.

What in particular are we worried about here? Is this more in theory than in our practical use case? 

I.e., if we were to have a number of steps and one of them calls useMakeStepActive, then maybe the step it was called from could have isCompleteCallback data it is handling that never gets handled?

Reply:
It could cause a similar problem to the issue described here: https://github.com/Automattic/payments-shilling/issues/3559

If you were on the contact details step and some code used this hook to make the payment method step active, the contact step would be made inactive but any changes to the contact details would not be saved.

This effectively bypasses the systems in place to make sure you always complete steps in the right order.